src/odemis/gui/comp/overlay/polygon.py

```diff
@@ -0,0 +1,217 @@
+
+    def on_left_down(self, evt):
+        if self.active.value and self.selected.value:
+            LineEditingMixin._on_left_down(self, evt)
+            self.cnvs.update_drawing()
+        else:
+            WorldOverlay.on_left_down(self, evt)
+
+    def on_left_up(self, evt):
+        if self.active.value:
+            LineEditingMixin._on_left_up(self, evt)
+            if self.finished_click_mixin:
+                self._phys_to_view()
+                offset = self.cnvs.get_half_buffer_size()
+                p_point = Vec(self.cnvs.view_to_phys(evt.Position, offset))
+                selected = self.is_point_in_shape(p_point)
+                self.selected._set_value(selected, must_notify=True)
+            self.cnvs.update_drawing()
+        WorldOverlay.on_left_up(self, evt)
+
+    def on_right_down(self, evt):
+        if self.active.value:
+            LineEditingMixin._on_right_down(self, evt)
+            self.cnvs.update_drawing()
+        else:
+            WorldOverlay.on_right_down(self, evt)
+
+    def on_right_up(self, evt):
+        if self.active.value:
+            LineEditingMixin._on_right_up(self, evt)
+            self._view_to_phys()
+            if len(self._points) <= 2:
+                logging.warning("Cannot create a polygon for less than 3 points.")
+                self.reset_click_mixin()
+                self._points.clear()
+            # FastEMROA.get_poly_field_indices expects list of nested tuples (y, x)
+            self.points.value = [(y, x) for x, y in self._points]
+            self.cnvs.update_drawing()
+        else:
+            WorldOverlay.on_right_up(self, evt)
+
+    def on_enter(self, evt):
+        if self.active.value:
+            self.cnvs.set_default_cursor(wx.CURSOR_CROSS)
+        else:
+            WorldOverlay.on_enter(self, evt)
+
+    def on_leave(self, evt):
+        if self.active.value:
+            self.cnvs.reset_default_cursor()
+        else:
+            WorldOverlay.on_leave(self, evt)
+
+    def on_motion(self, evt):
+        if self.active.value and self.selected.value:
+            LineEditingMixin._on_motion(self, evt)
+            if not self.dragging:
+                if self.hover == gui.HOVER_SELECTION:
+                    self.cnvs.set_dynamic_cursor(gui.DRAG_CURSOR)
+                elif self.hover == gui.HOVER_EDGE:
+                    self.cnvs.set_dynamic_cursor(wx.CURSOR_SIZING)
+                else:
+                    self.cnvs.reset_dynamic_cursor()
+            else:
+                self._view_to_phys()
+            self.cnvs.update_drawing()
+        else:
+            WorldOverlay.on_motion(self, evt)
+
+    def draw(self, ctx, shift=(0, 0), scale=1.0, line_width=4, dash=True):
+        """Draw the selection as a polygon"""
+        if self._points:
+            offset = self.cnvs.get_half_buffer_size()
+
+            # draws the dotted line
+            line_width = 5 if (self.active.value and self.selected.value) else 2
```

Comment:
Since you also use this line thickness in another place I think it would be nice to define it as a constant somewhere.